amm/src/main/scala/ammonite/interp/Preprocessor.scala

```diff
@@ -68,10 +70,21 @@ object Preprocessor{
 
         // comment holds comments or empty lines above the code which is not caught along with code
         for( (comment, code) <- s.value){
-          val ncomment = comment + "\n"*offset
 
-          // 1 is added as Separator parser eats up the '\n' following @
-          offset = offset + comment.count(_ == '\n') + code.map(_.count(_ == '\n')).sum + 1
+          //ncomment has required number of newLines appended based on OS and offset
+          //fastparse behaves differently for "\n" and "\r\n"
+          // thats why the `comment.substring(1)` thing is necessary
```

Comment:
This comment is an insufficient explanation of what is going on. You will need to elaborate on it to make it understandable